### PR TITLE
Fix double-free of path string in S3Client methods on error

### PR DESCRIPTION
**Pull request by robobun**
## What does this PR do?

Fixes a crash (double-free of the path string) in `S3Client` / `Bun.s3` methods when they throw after constructing the internal S3 blob.

### Reproduction

```js
// with no S3 credentials in the environment
try { Bun.S3Client.presign("some/key.txt"); } catch {}
try { Bun.S3Client.presign("some/key.txt"); } catch {}
Bun.gc(true);
```

crashes with:
- `panic: reached unreachable code` (debug), or
- `ASSERTION FAILED: wasRemoved` in `AtomStringImpl.cpp`

### Root cause

The static `Bun.S3Client.{presign,exists,size,stat,write,unlink}` and the corresponding `S3Client` instance methods all follow this pattern:

```zig
var path = try jsc.Node.PathLike.fromJS(globalThis, &args) orelse ...;
errdefer path.deinit();
...
var blob = try S3File.constructS3FileWithS3CredentialsAndOptions(globalThis, path, ...);
defer blob.detach();
return S3File.getPresignUrlFrom(&blob, globalThis, options); // can throw
```

`constructS3FileWithS3CredentialsAndOptions` / `constructS3FileInternalStore` transfer ownership of `path` into the blob's store (via `Blob.Store.initS3`, whose `path.toThreadSafe()` does not add a reference). If the subsequent operation throws, the `defer blob.detach()` frees the store — including the path — and then the outer `errdefer path.deinit()` frees it again.

### Fix

After the blob is successfully constructed, overwrite the caller's `path` / `path_or_blob` with an empty value so the `errdefer` becomes a no-op for subsequent errors. This matches the existing pattern in `Blob.zig` (e.g. `path_or_fd.* = .{ .path = .{ .string = bun.PathString.empty } }` after transferring ownership to `Blob.Store.initS3`).

## How did you verify your code works?

Added `test/js/bun/s3/s3-path-double-free.test.ts` which exercises all affected static and instance methods on the error path, forces GC, and asserts a clean exit. The test SIGABRTs on current main and passes with this fix.

### CODE REVIEW

**Comment by robobun** [comment]
<div><sup>Updated 3:48 PM PT - May 10th, 2026</sup></div>

@robobun, your commit dc475c87cebb3ae7f639e97bb05b454b078fe8b7 is building: [`#53190`](https://buildkite.com/bun/bun/builds/53190)
